mesheryctl/internal/cli/root/system/context/context_test.go

```diff
@@ -0,0 +1,364 @@
+package context
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	"github.com/layer5io/meshery/mesheryctl/internal/cli/root/config"
+	"github.com/layer5io/meshery/mesheryctl/pkg/utils"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+)
+
+var b *bytes.Buffer
+var update = flag.Bool("update", false, "update golden files")
+
+type CmdTestInput struct {
+	Name             string
+	Args             []string
+	ExpectedResponse string
+}
+
+func SetupContextEnv(t *testing.T, testfile string) {
+	path, err := os.Getwd()
+	if err != nil {
+		t.Error("unable to locate meshery directory")
+	}
+	viper.Reset()
+	viper.SetConfigFile(path + testfile)
+	err = viper.ReadInConfig()
+	if err != nil {
+		t.Errorf("unable to read configuration from %v, %v", viper.ConfigFileUsed(), err.Error())
+	}
+
+	configuration, err = config.GetMesheryCtl(viper.GetViper())
+	if err != nil {
+		t.Error("error processing config", err)
+	}
+}
+func SetupFunc(t *testing.T) {
+	b = bytes.NewBufferString("")
+	logrus.SetOutput(b)
+	utils.SetupLogrusFormatter()
+	ContextCmd.SetOut(b)
+}
```

Comment:
Shift this function to `utils/testing.go` change the title to be 
SetupLogrusGrabTesting(t *testing.T) bytes.Buffer {}

Eventually, this will get highly repeated to best to keep it central.

Reply:
@piyushsingariya  Made all the asked change.